barretenberg/cpp/src/barretenberg/stdlib/hash/poseidon2/sponge/sponge.hpp

```diff
@@ -53,16 +53,16 @@ template <size_t rate, size_t capacity, size_t t, typename Permutation, typename
         : builder(&builder_)
     {
         for (size_t i = 0; i < rate; ++i) {
-            state[i] = witness_t<Builder>(builder, 0);
+            state[i] = witness_t<Builder>::create_constant_witness(builder, 0);
         }
-        state[rate] = witness_t<Builder>(builder, domain_iv.get_value());
+        state[rate] = witness_t<Builder>::create_constant_witness(builder, domain_iv.get_value());
```

Comment:
this .get_value seems a little suspicious. Should we just change the argument to be a bb::fr instead so its explicitly a value?

Reply:
My analyzer didn't noticed this variable as dangerous...

Reply:
After fix

Reply:
Seem's like we have created constraint on domain_iv.get_value() as constant value, and after that there were created constraints for this variable in permutation

Reply:
yea, I'm not saying its dangerous in our current code but it looks a little weird to be passing in a field_t<Builder> when we just always pass in a bb::fr?